### PR TITLE
Activator usings replaced with CompiledLambdaActivatorFactory.

### PR DESCRIPTION
**Pull request by roman-kozachenko**

### CODE REVIEW

**Comment by Ivan-Razhnou** [comment]
Cool

File: src/msgpack/CompiledLambdaActivatorFactory.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Linq.Expressions;
+#if DNXCORE50
```
Comment (aensidhe) [inline src/msgpack/CompiledLambdaActivatorFactory.cs]: It's better to supress warning of Resharper by comment. I dislike compilation directives

File: src/msgpack/ArrayConverter.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ private TArray ReadArray(IMsgPackReader reader, MsgPackContext context, uint len
             uint length,
             IMsgPackConverter<TElement> converter)
         {
-            var array = creator == null ? (TArray)Activator.CreateInstance(typeof(TArray)) : creator();
+            var array = creator == null ? context.GetObjectActivator<TArray>(typeof (TArray))() : creator();
```
Comment (aensidhe) [inline src/msgpack/ArrayConverter.cs]: You should either pass type as parameter or use it in generic parameter.

File: src/msgpack/project.json [hunk 3]
```diff
@@ -9,24 +9,25 @@
...
-                "System.Collections.Concurrent": "4.0.11-beta-23516",
-                "System.IO": "4.0.11-beta-23516",
-                "System.Linq": "4.0.1-beta-23516",
-                "System.Reflection.TypeExtensions": "4.1.0-beta-23516",
-                "System.Runtime": "4.0.21-beta-23516",
-                "System.Runtime.Extensions": "4.0.11-beta-23516",
-                "System.Runtime.InteropServices": "4.0.21-beta-23516",
-                "System.Runtime.Serialization.Primitives": "4.0.11-beta-23409"
-            }
+          "dependencies": {
```
Comment (aensidhe) [inline src/msgpack/project.json]: We should sync our style settings for json.

File: tests/msgpack.tests/ObjectCreatorPerformanceTest.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+using Xunit;
+
+namespace TarantoolDnx.MsgPack.Tests
+{
+    public class ObjectCreatorPerformanceTest
+    {
+        private const int TestsCount = 10000000;
+
+        [Fact]
+        public void TestActivator()
```
Comment (aensidhe) [inline tests/msgpack.tests/ObjectCreatorPerformanceTest.cs]: This perfomance test is bad. We just slow down build and don't get any useful info

File: src/msgpack/CompiledLambdaActivatorFactory.cs (hunk 1 above)
Comment (roman-kozachenko) [inline src/msgpack/CompiledLambdaActivatorFactory.cs]: Well, good point. Replaced with comment.

File: src/msgpack/ArrayConverter.cs (hunk 2 above)
Comment (roman-kozachenko) [inline src/msgpack/ArrayConverter.cs]: I was confused by code in TryGenerateMapConverter, but you are right, single generic parameter will be enough.

File: tests/msgpack.tests/ObjectCreatorPerformanceTest.cs (hunk 4 above)
Comment (roman-kozachenko) [inline tests/msgpack.tests/ObjectCreatorPerformanceTest.cs]: Ok, removed it.

File: src/msgpack/project.json (hunk 3 above)
Comment (roman-kozachenko) [inline src/msgpack/project.json]: Changed tab size to 4 in VS options.

**Comment by aensidhe** [comment]
LGTM

**Pull request merged**